packages/commonwealth/client/scripts/controllers/app/login.ts

```diff
@@ -333,12 +335,21 @@ export async function startLoginWithMagicLink({
       walletSsoSource: WalletSsoSource.Email,
     });
 
+    return { bearer, address };
+  } else if (provider === WalletSsoSource.Farcaster) {
+    const bearer = await magic.farcaster.login();
+
+    const { address } = await handleSocialLoginCallback({
+      bearer,
+      walletSsoSource: WalletSsoSource.Farcaster,
+    });
+
```

Comment:
do we need to strictly follow this? can we reuse the `else` block logic below? That would redirect to `/finish_social_login` and then `handleSocialLoginCallback` is already called there along with some app init settings as well

Reply:
It's a little bit confusing but farcaster is not technically a social login. This is social logins require to use the webhook callback which farcaster does not use since it is decentralized. There is no entity (Like google or whatever) to submit the callback webhook for us.

This basically takes the same path as email login, the only difference is that it needs to call the magic farcaster login method